### ISSUE TITLE
Segmentation fault on running cifar10-resnet.py

### ISSUE THREAD

**Issue by jyh2986** [ppwwyyxx/tensorpack#237]
When I run cifar10-resnet.py in example, I got a segmentation fault.
Tensorflow is 1.0.1 and Tensorpack is 0.1.8.

I have been used Tensorpack 1.0 & tensorflow 0.12.1, and I upgraded both to the latest version.
In old versions, There was no problems.

The error occurred in ModelSaver(). Just commenting that line, it is ok.
I think it is related with saving summaries or ckpt files.

I'm using docker, and I didn't re-install opencv after upgrading.

**Comment by ppwwyyxx** [comment]
Sorry but there's nothing I can do about it given the little information.
In most cases, any segfault is a problem of either tensorflow or the environment.

It's better if you could give more information, e.g., when does it segfault. Does it always segfault at the same place. At which line does it segfault. Have you tried other TF versions (e.g. TF1.1). Have you tried disable/enable GPU. Have you tried other machines, etc...

**Comment by jyh2986** [comment]
I'm sorry I gave you a little clue.
Same problem in other machines (I used same docker env.)

I thought that the segfault is always occurred on ModelSaver, but this is not true.
If I add meaningless code(e.g. add dummy argument), it stops immediately like belows.

[0421 09:09:53 @base.py:116] Setup callbacks graph ...
[0421 09:09:53 @base.py:187] Building predictor graph towerp0 on gpu=0 ...
Segmentation fault (core dumped)

One of the other phenomena is that segfault always occurs at the end, even if it is not interrupted.

I think I have to find more clues to explain you.
One of the suspects is that I didn't install TF & TP from scratch.

**Comment by ppwwyyxx** [comment]
Have you tried to disable GPU? How did you install opencv? Opencv can cause segfault if installed from 3rd-party prebuilt or sometimes on GPU.

**Comment by jyh2986** [comment]
This is the failure procedures.
1. Install tensorflow 0.12.1 using docker and installed opencv(make install), tensorpack
  - There's no problem.
2. Upgrade tensorflow using "pip install -upgrade" to v1.0.1, and tensorpack 0.1.8
  - Segfault occurred. 
3. I removed opencv and make install again.
  - Segfault occurred.


After that, I re-installed tensorflow(using docker), opencv and tensorpack from scratch. And there's no problem. It might be caused by upgrading tensorflow.
I tried to identify the problem with existing repository, but it failed.

Anyway, it is solved.
Thank you.
